lang/cs/Org.Apache.REEF.Wake/Remote/ListTcpPortProvider.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Wake.Remote.Parameters;
+
+namespace Org.Apache.REEF.Wake.Remote
+{
+    public sealed class ListTcpPortProvider : ITcpPortProvider
+     {
+        private readonly ISet<int> _tcpPortList;
+
+        [Inject]
+        public ListTcpPortProvider([Parameter(typeof(TcpPortList))] ISet<int> tcpPortList)
+        {
+            _tcpPortList = tcpPortList;
+        }
+
+        IEnumerator<int> IEnumerable<int>.GetEnumerator()
+        {
```

Comment:
why do we need different Enumerators here?

Reply:
Added it since compiler requires these methods as the class implements IEnumerable<int>.

---
In reply to: [200423311](https://github.com/sharathmalladi/reef/pull/6#discussion_r200423311) [](ancestors = 200423311)